src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -136,14 +83,13 @@ static MsQuicApi()
                 {
                     if (NativeLibrary.TryGetExport(msQuicHandle, "MsQuicOpenVersion", out IntPtr msQuicOpenVersionAddress))
                     {
-                        NativeApi* vtable;
-                        delegate* unmanaged[Cdecl]<uint, NativeApi**, uint> msQuicOpenVersion =
-                            (delegate* unmanaged[Cdecl]<uint, NativeApi**, uint>)msQuicOpenVersionAddress;
-                        uint status = msQuicOpenVersion(MsQuicVersion, &vtable);
-                        if (MsQuicStatusHelper.SuccessfulStatusCode(status))
+                        QUIC_API_TABLE* apiTable;
+                        delegate* unmanaged[Cdecl]<uint, QUIC_API_TABLE**, int> msQuicOpenVersion = (delegate* unmanaged[Cdecl]<uint, QUIC_API_TABLE**, int>)msQuicOpenVersionAddress;
+                        int status = msQuicOpenVersion(MsQuicVersion, &apiTable);
+                        if (StatusSucceeded(status))
                         {
                             IsQuicSupported = true;
-                            Api = new MsQuicApi(vtable);
+                            Api = new MsQuicApi(apiTable);
```

Comment:
I wonder if you should add a check to query [QUIC_PARAM_GLOBAL_LIBRARY_VERSION](https://github.com/microsoft/msquic/blob/main/src/cs/lib/msquic_generated.cs#L2542) so you may specify a minimum Minor version number in the future. Right now you're updating to 2.0, but say 2.1 adds a new feature that you take a dependency on; then you will only be able to use 2.1+, and not 2.0. So you don't strictly need it yet, but it's probably a good thing to add now, while we're thinking about it.